### PR TITLE
(fix) Controller I/O table: sort Action column by display string

### LINKED ISSUE #13036: MIDI table: order of rows wonky
### Bug Description

A strange sort order;
image

"Actions are sorted by control, and IIRC that is
D ouble
F our thirds
H alf
T hree fourths
T wo thirds etc"

#12934 

### Version

2.5-alpha

### OS

_No response_

### PR DESCRIPTION
**Pull request by ronso0**
Addresses an old TODO and fixes #13036

### CODE REVIEW

**Review by Swiftb0y — APPROVED** [review]

File: src/controllers/controllerinputmappingtablemodel.cpp
Comment (ronso0) [reply src/controllers/controllerinputmappingtablemodel.cpp:222]: Ah okay, you think clang expects all returns of a case to be indented identically?
I'll try that, but tbh I'm a bit tired of making clang-format happy with previously unrequired formatting.

Comment (Swiftb0y) [reply src/controllers/controllerinputmappingtablemodel.cpp:222]: > Ah okay, you think clang expects all returns of a case to be indented identically?

Nah, the syntax rules about switch-case is just a bit weird in general, so I just think this is bug in clang format which might be worked around that way. 

>  I'll try that, but tbh I'm a bit tired of making clang-format happy with previously unrequired formatting.

I agree. I mean I wouldn't want to go without clang-format though so I usually just do what makes it happy, even if I don't agree with it.

Comment (Swiftb0y) [reply src/controllers/controllerinputmappingtablemodel.cpp:222]: Whatever it takes, just make it happy even if its accepting the garbage it produces. Introducing a CI failure deliberately is not worth the potential hassle later.

Comment (ronso0) [reply src/controllers/controllerinputmappingtablemodel.cpp]: :+1:

**Comment by ronso0** [comment]
Merge?

**Comment by Swiftb0y** [comment]
I'm sorry I don't mean to reiterate, but I think we should just accept the bogus clang-format suggestion because I don't want to merge this PR unless its green. Merging a failing PR would probably have annoying effects later (much more than a misformatted line). So can you please apply the pre-commit patch, amend and force push?

**Comment by ronso0** [comment]
Oh, sorry.
Locally it passed with the extra {} so I didn't check CI.
But here clang insist on the indentation :shrug: 

Will fix it.

**Comment by daschuer** [comment]
@Swiftb0y This looks good now, merge?

**Comment by ronso0** [comment]
_friendly pinging @Swiftb0y_

**Review by daschuer — APPROVED** [review]
I think we can merge this now. @Swiftb0y If you find time for a review. We may fix the findings after merge.

**Comment by Swiftb0y** [comment]
sorry for not responding earlier, this got lost among the flood of notifications. lgtm indeed.

**Pull request merged**
